### PR TITLE
web: add keyboard shortcuts for snapshot, endpoints

### PR DESCRIPTION
**Pull request by nicks**
Hello @hyu, @maiamcc,

Please review the following commits I made in branch nicks/morenav2:

28d0405a9b002d826c833dce7c00692cb80ceb29 (2020-10-07 12:03:39 -0400)
web: add keyboard shortcuts for snapshot, endpoints

ed79dfa44bb185a638e5e72fbf086e850ffd9a00 (2020-10-07 09:57:55 -0400)
web: add navigate with 1, 2, 3 keys

Code review reminders, by giving a LGTM you attest that:

* Commits are adequately tested
* Code is easy to understand and conforms to style guides
* Incomplete code is marked with TODOs
* Code is suitably instrumented with logging and metrics

### CODE REVIEW

File: web/src/ResourceInfoKeyboardShortcuts.test.tsx [hunk 1]
```diff
@@ -0,0 +1,53 @@
...
+})
+it("two endpoint urls trigger second", () => {
+  shortcuts([
+    { url: "https://tilt.dev:4000" },
+    { url: "https://tilt.dev:4001" },
+  ])
+  fireEvent.keyDown(document.body, { keyCode: numKeyCode(2), shiftKey: true })
+  expect(endpointUrl).toEqual("https://tilt.dev:4001")
+})
+it("handles snapshot shortcut", () => {
```
Comment (maiamcc) [inline web/src/ResourceInfoKeyboardShortcuts.test.tsx:48]: nbd but consider a test for:`showSnapshotButton` is false, `s` does nothing

**Review by maiamcc — APPROVED** [review]
🙌

File: web/src/ResourceInfo.tsx [hunk 2]
```diff
@@ -91,6 +94,15 @@ let SnapshotButtonSvg = styled(SnapshotSvg)`
   margin-right: ${s.SizeUnit(0.25)};
 `
 
+// TODO(nick): Put this in a global React Context object with
+// other page-level stuffs
+function openEndpointUrl(url: string) {
+  // We deliberately don't use rel=noopener. These are trusted tabs, and we want
```
[the remark this answers is not in the document]
Comment (nicks) [reply web/src/ResourceInfo.tsx:100]: yes, rel=noopener is for a very narrow (but common) threat model

https://owasp.org/www-community/attacks/Reverse_Tabnabbing

where
1) Alice hosts a platform of user-generated content
2) Evil Eve authors some content on Alice's site
3) Bob logs into Alice's site to read content

and Evil Eve is trying to trick Bob into giving them a secret that's meant for Alice

this is nowhere near the threat model for the tilt local UI

File: web/src/ResourceInfoKeyboardShortcuts.tsx [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+      if (!endpoint || !endpoint.url) {
+        return
+      }
+
+      this.props.openEndpointUrl(endpoint.url)
+      e.preventDefault()
+      return
+    }
+
+    if (e.metaKey || e.altKey || e.ctrlKey || e.isComposing) {
```
[the remark this answers is not in the document]
Comment (nicks) [reply web/src/ResourceInfoKeyboardShortcuts.tsx]: done

File: web/src/SecondaryNavKeyboardShortcuts.tsx [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+
+type Props = {
+  logUrl: string
+  alertsUrl: string
+  facetsUrl: string | null
+  history: History
+}
+
+/**
+ * Sets up keyboard shortcuts that depend on the state of the sidebar.
```
[the remark this answers is not in the document]
Comment (nicks) [reply web/src/SecondaryNavKeyboardShortcuts.tsx]: done

File: web/src/ResourceInfoKeyboardShortcuts.test.tsx (hunk 1 above)
Comment (nicks) [reply web/src/ResourceInfoKeyboardShortcuts.test.tsx:58]: done

**Pull request merged**